### ISSUE TITLE
downstream-logging-relation-changed failed when Loki is using self signed certs

### ISSUE THREAD

**Issue by peppepetra** [canonical/cos-proxy-operator#110]
### Bug Description

cos-proxy should provide an option similar to grafana-agent to skip the verification for insecure TLS, otherwise the relation with Loki fails when cos-lite is deployed with the provided [tls-overlay](https://github.com/canonical/cos-lite-bundle/blob/main/overlays/tls-overlay.yaml).

### To Reproduce

1. juju deploy ./cos_lite.yaml --trust --overlay ./offers_overlay.yaml --overlay ./storage_overlay.yaml --overlay tls-overlay.yaml
2. juju consume cos-microk8s-localhost:admin/cos.loki-logging cos-loki-logging
3. juju deploy cos-proxy
4. juju add-relation cos-proxy:downstream-logging cos-loki-logging:logging

### Environment

cos-proxy rev. 52 channel edge 

### Relevant log output

```shell
Traceback (most recent call last):
  File "./src/charm.py", line 519, in <module>
    main(COSProxyCharm)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/main.py", line 436, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/main.py", line 144, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/framework.py", line 351, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/framework.py", line 853, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/framework.py", line 942, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/lib/charms/vector/v0/vector.py", line 227, in _on_log_relation_changed
    self.on.config_changed.emit(config=self.config)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/framework.py", line 351, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/framework.py", line 853, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-cos-proxy-0/charm/venv/ops/framework.py", line 942, in _reemit
    custom_handler(event)
  File "./src/charm.py", line 395, in _write_vector_config
    r = request.urlopen(dest)
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 1397, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/usr/lib/python3.8/urllib/request.py", line 1357, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)>
```

**Comment by lucabello** [comment]
@simskij Do we want the cos-proxy charm to be able to "skip verify" or do we want to the charm to trust the self-signed certificate?

**Comment by sed-i** [comment]
Duplicate of #111 and is no longer a problem after #118 merged.
Closing.
